src/hexer_docs.erl

```diff
@@ -0,0 +1,106 @@
+-module(hexer_docs).
+
+-export([publish/0]).
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+-spec publish() -> ok | {error, any()}.
+publish() ->
+  hexer_utils:print("Publishing..."),
+  %% We need the following information to build the package:
+  %% - App Directory
+  %% - Name
+  %% - Version
+  %% - AppDetails from *.app.src
+  %% - Deps
+
+  %% We assume the app's dir is the current directory.
+  AppDir = ".",
+  #{ name    := Name
+   , version := Version
+   } = load_app_info(),
+  publish(AppDir, Name, Version).
+
+%%------------------------------------------------------------------------------
+%% Internal Functions
+%%------------------------------------------------------------------------------
+-type details() :: #{ description  => string()
+                    , applications => [atom()]
+                    , maintainers  => [string()]
+                    , licenses     => [string()]
+                    , files        => [binary()] %% Files to be included
+                    , links        => [{Label :: string(), Label :: string()}]
+                    }.
+
+-type app_info() :: #{ name    => atom()
+                     , version => string()
+                     , details => details()
+                     }.
+
+-spec load_app_info() -> app_info().
+load_app_info() ->
+  case hexer_utils:find_single_file(["ebin/*.app", "src/*.app.src"]) of
+    {ok, Path} ->
+      {ok, [AppSrc]} = file:consult(Path),
+      {application, Name, Details} = AppSrc,
+      DetailsMap = maps:from_list(Details),
+      VSN = maps:get(vsn, DetailsMap),
+      Version = transform_vsn_git_to_tag(VSN),
+      #{ name    => Name
+       , version => Version
+       };
+    notfound ->
+      throw(app_file_not_found)
+  end.
+
+-spec transform_vsn_git_to_tag(any()) -> any().
+transform_vsn_git_to_tag(git) ->
```

Comment:
This is copied and pasted from another module in this repo. It should be moved to a common place, for instance `hexer_utils`